ios/keyman/Keyman/Keyman/MainViewController.swift

```diff
@@ -0,0 +1,1200 @@
+    let vAdjLscpe: CGFloat = -1.6
+
+    let vAdj = UIInterfaceOrientationIsPortrait(orientation) ? vAdjPort : vAdjLscpe
+    let navBarHeight = self.navBarHeight()
+    let y = (navBarHeight - icon.size.height) / 2.0 + vAdj
+
+    let customButton = UIButton(type: .custom)
+    customButton.setImage(icon, for: .normal)
+    customButton.setImage(iconHighlighted, for: .highlighted)
+    customButton.frame = CGRect(x: 0.0, y: y, width: icon.size.width, height: icon.size.height)
+    customButton.addTarget(self, action: selector, for: .touchUpInside)
+
+    let containerView = UIView(frame: CGRect(x: 0.0, y: 0.0, width: icon.size.width, height: navBarHeight))
+    containerView.backgroundColor = UIColor.clear
+    containerView.addSubview(customButton)
+    return UIBarButtonItem(customView: containerView)
+  }
+
+  override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+
+    if wasKeyboardVisible {
+      textView.becomeFirstResponder()
+    }
+
+    if !textView.isEditable {
+      showActivityIndicator()
+    } else if shouldShowGetStarted() {
+      perform(#selector(self.showGetStartedView), with: nil, afterDelay: 1.0)
+    }
+  }
+
+  deinit {
+    NotificationCenter.default.removeObserver(self)
+  }
+
+  override func willRotate(to toInterfaceOrientation: UIInterfaceOrientation, duration: TimeInterval) {
+    let orientation: UIInterfaceOrientation = toInterfaceOrientation
+    for view in overlayWindow.subviews {
+      UIView.animate(withDuration: duration, delay: 0.0, options: ([]), animations: {() -> Void in
+        view.transform = self.transform(for: orientation)
+      }, completion: nil)
+    }
+
+    // TODO: Refactor
+    if UIDevice.current.userInterfaceIdiom == .phone {
+      let screenRect = UIScreen.main.bounds
+      let size = CGFloat.maximum(screenRect.width, screenRect.height)
+      if size > 568.0 {
+        // Navbar for iPhone 6 & 6 Plus
+        let image: UIImage
+        if UIInterfaceOrientationIsPortrait(orientation) {
+          image = #imageLiteral(resourceName: "navbar-Portrait.png")
+        } else {
+          image = #imageLiteral(resourceName: "navbar-Landscape-568h.png")
+        }
+        let bgImg = image.resizableImage(withCapInsets: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0),
+                                         resizingMode: .stretch)
+        navigationController?.navigationBar.setBackgroundImage(bgImg, for: UIBarMetrics.default)
+      }
+    } else {
+      if UIInterfaceOrientationIsPortrait(orientation) {
+        navigationController?.navigationBar.setBackgroundImage(#imageLiteral(resourceName: "navbar-Portrait.png"), for: UIBarMetrics.default)
+      } else {
+        navigationController?.navigationBar.setBackgroundImage(#imageLiteral(resourceName: "navbar-Landscape.png"), for: UIBarMetrics.default)
+      }
+    }
+    popover?.dismiss(animated: false)
+    _ = dismissDropDownMenu()
+  }
```

Comment:
Hmm, the Obj-C line

`[actionSheet dismissWithClickedButtonIndex:actionSheet.cancelButtonIndex animated:NO];`

seems to have no equivalent here.  Why is that?

Reply:
I should have commented on this PR earlier.

`actionSheet` was previously only set for iOS < 8:
https://github.com/keymanapp/keyman/blob/2b841e5e82831535c31d0378ace34484f921118b/ios/keyman/Keyman/Keyman/MainViewController.m#L784
https://github.com/keymanapp/keyman/blob/2b841e5e82831535c31d0378ace34484f921118b/ios/keyman/Keyman/Keyman/MainViewController.m#L862

and when `UIAlertController` is not available ([also iOS < 8](https://developer.apple.com/documentation/uikit/uialertcontroller)):
https://github.com/keymanapp/keyman/blob/2b841e5e82831535c31d0378ace34484f921118b/ios/keyman/Keyman/Keyman/MainViewController.m#L901
https://github.com/keymanapp/keyman/blob/2b841e5e82831535c31d0378ace34484f921118b/ios/keyman/Keyman/Keyman/MainViewController.m#L928

So I just removed `actionSheet` and related methods.